### PR TITLE
Quick UX fixes

### LINKED ISSUE #3623: Quick UX wins for the next release
## Feature description

This is about some low hanging fruits that came up during UX reviews of the Stories editor. I figured it would be nice to address this for this release as it would result in some nice little improvements.

See #3611 for details and screenshots

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Do not display a featured image notice in the document sidebar  
  -> It should still be displayed in the pre-publish panel
* Opacity controls should be hidden when no color has been selected yet  
  -> changing the opacity has no effect otherwise
* When setting the background color for a page with a background media item, it should do so with reduced opacity. 
* Labelling of color panels should be more aligned between pages and text elements
* When editing a text element, the color panel should initially be open

## Implementation brief

* In `BackgroundColorSettings`, when adding a color for the first time and there is a media item, set `overlayOpacity` to `50`
* Change labels of color settings panels to make them more aligned
* Remove `initialOpen` prop of text element color panel
* Only display opacity controls when there are colors set
* Remove `withStoryFeaturedImageNotice` HOC and its usage

## QA testing instructions

1. Create a new story
1. **Verify:** No featured image warning should be shown on the right
1. Set a background image on the page
1. **Verify:** There should be no opacity control on the right side
1. Set a background color on the page
1. **Verify:** The background color should have opacity so that the image is still visible
1. **Verify:** There should now be a opacity control on the right side, and opacity can be changed at will
1. Add a text element
1. **Verify:** The color panel on the right should be open
1. **Verify:** There should be no opacity control on the right side
1. Set a background color on the text element
1. **Verify:** There should now be a opacity control on the right side, and opacity can be changed at will

## Demo

Remove annoying featured image notice in the document sidebar

Screenshot 2019-10-24 at 23 08 53

Hide opacity controls when no colors have been added yet

No color:

Screenshot 2019-10-24 at 23 09 44

Color:

Screenshot 2019-10-24 at 23 09 53

Page: when adding the first color, reduce opacity in order to not cover the background media

The image is still visible after adding a color:

Screenshot 2019-10-24 at 23 08 29

Initially open the color settings panel (currently it's collapsed by default)

Screenshot 2019-10-24 at 23 11 10

Make color panels more aligned between pages and text elements

Page:

Screenshot 2019-10-24 at 23 07 53

Text block:

Screenshot 2019-10-24 at 23 07 43

## Changelog entry

* Fixed a few UX inconsistencies in the editor

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Fixes #3623.

This PR attempts to fix a few UX issues here and there:

* [x] Remove annoying featured image notice in the document sidebar
* [x] Hide opacity controls when no colors have been added yet
* [x] Page: when adding the first color, reduce opacity in order to not cover the background media
* [x] Initially open the color settings panel (currently it's collapsed by default)
* [x] Make color panels more aligned between pages and text elements

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by swissspidy** [comment]
> Remove annoying featured image notice in the document sidebar

Screenshot 2019-10-24 at 23 08 53

> Hide opacity controls when no colors have been added yet

No color:

Screenshot 2019-10-24 at 23 09 44


Color:

Screenshot 2019-10-24 at 23 09 53

> Page: when adding the first color, reduce opacity in order to not cover the background media

The image is still visible after adding a color:

Screenshot 2019-10-24 at 23 08 29

> Initially open the color settings panel (currently it's collapsed by default)

Screenshot 2019-10-24 at 23 11 10

> Make color panels more aligned between pages and text elements

Page:

Screenshot 2019-10-24 at 23 07 53

Text block:

Screenshot 2019-10-24 at 23 07 43

**Comment by swissspidy** [comment]
@samitron7 @jasti WDYT about addressing these low hanging fruits already?

**Comment by jasti** [comment]
Love it. Thanks for this PR!

File: assets/src/stories-editor/blocks/amp-story-page/background-color-settings.js [hunk 1]
```diff
@@ -23,9 +23,14 @@ import {
  * @param {Array} props.backgroundColors Current background colors.
  * @param {Function} props.setAttributes setAttributes callback.
  * @param {number} props.overlayOpacity Overlay opacity.
+ * @param {number} props.hasMedia Whether the page has a media item set.
  * @return {ReactElement} Component.
  */
-const BackgroundColorSettings = ( { backgroundColors, setAttributes, overlayOpacity } ) => {
+const BackgroundColorSettings = ( { backgroundColors, setAttributes, overlayOpacity, hasMedia } ) => {
+	const hasColors = backgroundColors
```
Comment (barklund) [inline assets/src/stories-editor/blocks/amp-story-page/background-color-settings.js]: Could be simplified to:

`const HasColors = backgroundColors.some ( ( color ) => Object.values( color ).some( Boolean ) );`

Comment (swissspidy) [reply assets/src/stories-editor/blocks/amp-story-page/background-color-settings.js]: Much better!

Comment (swissspidy) [reply assets/src/stories-editor/blocks/amp-story-page/background-color-settings.js]: Went with the first one but used `color && Object.values( color ).some( Boolean )` as `color` it's not guaranteed to be an object

**Pull request merged**